src/celeritas/optical/surface/model/DielectricInteractor.hh

```diff
@@ -0,0 +1,164 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/model/DielectricInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/random/distribution/BernoulliDistribution.hh"
+#include "celeritas/optical/CoreTrackView.hh"
+
+#include "FresnelCalculator.hh"
+#include "ReflectionFormSampler.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sample optical interactions for dielectric-dielectric and dielectric-metal
+ * interfaces.
+ *
+ * For both interfaces, the reflectivity is first calculated from Fresnel
+ * equations and sampled to determine if the photon will reflect or refract. If
+ * it reflects, then the UNIFIED model is used to handle the different forms of
+ * reflection. If it refracts, then dielectric-dielectric interfaces will use
+ * Snell's law to determine the refracted wave direction and polarization. For
+ * dielectric-metal interfaces, refracted waves are just absorbed.
+ */
+class DielectricInteractor
+{
+  public:
+    struct Builder
```

Comment:
I think for this to be called "Interactor::Builder" it would need to return an `Interactor` without changing its state. We can brainstorm different designs later, but for now it should be easy and obvious enough to rename it `Executor`.